src/commands/base.mjs

```diff
@@ -61,5 +61,93 @@ export class BaseCommand extends ShellRunner {
     this.chartManager = opts.chartManager
     this.configManager = opts.configManager
     this.depManager = opts.depManager
+    this._configMaps = new Map()
+  }
+
+  /**
+   * Dynamically builds a class with properties from the provided list of flags
+   * and extra properties, will keep track of which properties are used.  Call
+   * getUnusedConfigs() to get an array of unused properties.
+   *
+   * @param {string} configName the name of the configuration
+   * @param {CommandFlag[]} flags an array of flags
+   * @param {string[]} [extraProperties] an array of extra properties
+   * @returns {Object} the instance of the new class
+   */
+  getConfig (configName, flags, extraProperties = []) {
```

Comment:
FYI, Nathan and I reviewed this in a 1:1.  He is okay with how I'm using this here and what I'm using it for.  It is more of an exception though, and not necessarily to be used as a pattern.